File: Data Structures & Algorithms/valid-sudoku/submission-5.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:

        for i in range(len(board)):
            s = set()
            for j in range(len(board)):
                num = board[i][j]
                if num == ".":
                    continue
                if num in s:
                    return False
                s.add(num)
        
        for i in range(len(board)):
            s = set()
            for j in range(len(board)):
                num = board[j][i]
                if num == ".":
                    continue
                if num in s:
                    return False
                s.add(num)

        for i in range(0,9,3):
            for j in range(0,9,3):
                s = set()
                for r in range(i, i+3, 1):
                    for c in range(j, j+3, 1):
                        num = board[r][c]
                        if num == ".":
                            continue
                        if num in s:
                            return False
                        s.add(num)
        return True

File: Data Structures & Algorithms/string-encode-and-decode/submission-5.go
type Solution struct{}

func (s *Solution) Encode(strs []string) string {
    var res strings.Builder
    for _, str:= range strs{
        s := strconv.Itoa(len(str)) + "#" + str

        res.WriteString(s)
    }

    //fmt.Println(res.String())
    return res.String()
}

func (s *Solution) Decode(encoded string) []string {
    l,r := 0,0
    res := make([]string,0)
    for r < len(encoded) {
        for encoded[r] != '#'{
            r += 1
        }
        //fmt.Println(r)
        length,_ := strconv.Atoi(encoded[l:r])
        //fmt.Println(length)
        str := encoded[r+1:r + 1 + length]
        //fmt.Println(str)
        res = append(res, str)
        
        r += length + 1
        l = r
    }
    return res
}
